include/pluginlib/class_desc.h

```diff
@@ -1,84 +1,38 @@
-/*********************************************************************
-*
-* Software License Agreement (BSD License)
-*
-*  Copyright (c) 2008, Willow Garage, Inc.
-*  All rights reserved.
-*
-*  Redistribution and use in source and binary forms, with or without
-*  modification, are permitted provided that the following conditions
-*  are met:
-*
-*   * Redistributions of source code must retain the above copyright
-*     notice, this list of conditions and the following disclaimer.
-*   * Redistributions in binary form must reproduce the above
-*     copyright notice, this list of conditions and the following
-*     disclaimer in the documentation and/or other materials provided
-*     with the distribution.
-*   * Neither the name of Willow Garage, Inc. nor the names of its
-*     contributors may be used to endorse or promote products derived
-*     from this software without specific prior written permission.
-*
-*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
-*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
-*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
-*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
-*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
-*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
-*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
-*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
-*  POSSIBILITY OF SUCH DAMAGE.
-*
-*********************************************************************/
+/*
+ * Copyright (c) 2017, Open Source Robotics Foundation, Inc.
```

Comment:
our current copyright/licence checker will need the licence to be explicitly stated to match the licence template. we would need to add `Software License Agreement (BSD License 2.0)` between the `All rights reserved` line and and the body of the licence for it to work. (Assuming that `ament_copyright` eventually gets updated to handle lines with multiples leading whitespaces and block comments)

Reply:
I was gonna leave that until we are able to turn on the copyright linter for the melodic-devel branch (if we're ever able to do so). These files won't exist on the ros2 branch.

Reply:
Given that the files original files (with the Willow Garage copyright) have the licence listed explicitly and the files is ROS 2 packages have it as well, I would add it to these new files to stay consistent.
Then that won't be enough for it to pass the linter and I agree with you, we can address cosmetics later when/if we can enable the copyright linter

Reply:
I copied this from this file:

https://github.com/ros/pluginlib/blob/9e79885c2b4997e7b4abb3a3a607b5e098e0f469/include/pluginlib/class_loader.h#L1-L28

Also, these are new files, so that's why I didn't include the Willow Garage copyright.

What exactly do you want me to do here? I'm not sure I follow what you mean about explicitly listing the license.

Reply:
Oh looks like it was inconsistent in the first place, so that's fine, we can merge thisas is :+1:  and will iterate if we end up turning on the linter.

---
What I meant (for completeness) is that this file originally started with 
```
* Software License Agreement (BSD License)
 *
 *  Copyright (c) 2008, Willow Garage, Inc.
 *  All rights reserved.
```

and now has only:
```
+ * Copyright (c) 2017, Open Source Robotics Foundation, Inc.
+ * All rights reserved.
```
and it looks like the linter expects this additional line as part of the BSD header (https://github.com/ament/ament_lint/blob/767257ff141d7957f3c8c7b92106aced95794537/ament_copyright/ament_copyright/template/bsd2_header.txt)

Reply:
I see what you mean now. I added the explicit name in 7aeabc9, we'll still have to work on making this consistent when we turn on the linter.